runtime/bigarray.c

```diff
@@ -683,20 +684,23 @@ static intnat caml_ba_offset(struct caml_ba_array * b, intnat * index)
   int i;
 
   offset = 0;
-  if ((b->flags & CAML_BA_LAYOUT_MASK) == CAML_BA_C_LAYOUT) {
+  switch ((enum caml_ba_layout)(b->flags & CAML_BA_LAYOUT_MASK)) {
+  case CAML_BA_C_LAYOUT:
     /* C-style layout: row major, indices start at 0 */
     for (i = 0; i < b->num_dims; i++) {
       if ((uintnat) index[i] >= (uintnat) b->dim[i])
         caml_array_bound_error();
       offset = offset * b->dim[i] + index[i];
     }
-  } else {
+    break;
+  case CAML_BA_FORTRAN_LAYOUT:
     /* Fortran-style layout: column major, indices start at 1 */
     for (i = b->num_dims - 1; i >= 0; i--) {
       if ((uintnat) (index[i] - 1) >= (uintnat) b->dim[i])
         caml_array_bound_error();
       offset = offset * b->dim[i] + (index[i] - 1);
     }
+    break;
```

Comment:
Shouldn't this be here (cf. L1140)
```suggestion
    break;
  default: CAMLtrap();
```

Reply:
I've thought about this a bit more, and the bigarray layout is represented by a single bit (`#define CAML_BA_LAYOUT_MASK 0x100`). There are no invalid values here, if the compiler warns, we can mark it as unreachable (if ever we start using `-Wswitch-default` or `-Wswitch-enum`, but they can make the unnecessarily verbose).

Reply:
so I've removed the default case.

Reply:
Ah, the default case is needed otherwise the compiler thinks there are code paths leaving some variables uninitialized. I'll add `default: unreachable();` for switches on the bigarray layout.